tests/test_asgi.py

```diff
@@ -0,0 +1,262 @@
+from __future__ import annotations
+
+import asyncio
+import io
+import os
+import stat
+import tempfile
+
+import pytest
+
+from whitenoise.asgi import (
+    AsgiWhiteNoise,
+    convert_asgi_headers,
+    convert_wsgi_headers,
+    read_file,
+    receive_request,
+    serve_static_file,
+)
+from whitenoise.responders import StaticFile
+
+from .test_whitenoise import application as whitenoise_application
+from .test_whitenoise import files
+
+
+@pytest.fixture()
+def loop():
+    return asyncio.get_event_loop()
+
+
+class MockStat:
+    def __init__(self, st_mode, st_size, st_mtime):
+        self.st_mode = st_mode
+        self.st_size = st_size
+        self.st_mtime = st_mtime
+
+
+@pytest.fixture()
+def static_file_sample():
+    content = b"01234567890123456789"
+    modification_time = "Sun, 09 Sep 2001 01:46:40 GMT"
+    modification_epoch = 1000000000
+    temporary_file = tempfile.NamedTemporaryFile(suffix=".js", delete=False)
```

Comment:
use pytest's `tmp_path` fixture rather than writing manual file tempfile setup and cleanup logic

https://docs.pytest.org/en/latest/how-to/tmp_path.html